### ISSUE TITLE
Counter - registry.counter(metadata)- IllegalArgumentException

### ISSUE THREAD

**Issue by rsvoboda** [smallrye/smallrye-metrics#43]
Counter - registry.counter(metadata) returns IllegalArgumentException on second invocation of the endpoint. The same app works fine on Open Liberty.

JavaDoc for counter(Metadata metadata) says ` @return a new or pre-existing @link Counter`.
Second invocation should return the pre-existing counter.

Code:
```java
@GET
@Path("/stats")
public String getTotalDonations() {
    registry.counter(statsHitsCounterMetadata).inc();
    return "12345";
}
```

Error:
```
Caused by: java.lang.IllegalArgumentException:
  Previously registered metric statsHits was not flagged as reusable
```

OpenLiberty:
```json
curl http://localhost:9080/hello-prometheus/resources/coffee/stats
curl http://localhost:9080/hello-prometheus/resources/coffee/stats
curl http://localhost:9080/hello-prometheus/resources/coffee/stats

curl -k -u theUser:thePassword -H "Accept: application/json" https://localhost:9443/metrics/application/statsHits
{"statsHits":3}
```

https://github.com/rsvoboda/rsvoboda-playground/blob/master/microprofile-metrics-hello-prometheus/src/main/java/com/sebastian_daschner/hello_prometheus/CoffeesResource.java#L106

**Comment by jmartisk** [comment]
This was discussed on the spec call and there was an agreement that the 1.0 and 1.1 spec is not very clear about this. It was decided that in 2.0 spec, the default value of `reusable` attribute for metrics created from metadata will be `true` (now it is undefined by spec), hence this use case will work as described even without explicitly setting `reusable` to true. 

For backward compatibility, we decided to keep the behavior as-is in 1.1 spec and in SmallRye 1.1. For this use case to work with 1.1, you will need to explicitly set `reusable=true` in the metadata. I will prepare the changes in SmallRye and specification for 2.0.

**Comment by baranowb** [comment]
@jmartisk :
JDOC: The default is <tt>false</tt>, which prevents reusing.
Im not sure if its weld or small-rye that is at fault. In case of some metrics, small-rye creates OriginTrackedMetadata (https://github.com/smallrye/smallrye-metrics/blob/master/implementation/src/main/java/io/smallrye/metrics/interceptors/MetricsInterceptor.java#L84) which bypass the isReusable check at https://github.com/smallrye/smallrye-metrics/blob/master/implementation/src/main/java/io/smallrye/metrics/MetricsRegistryImpl.java#L214

In case of offending metric - the metadata is being created explicitly, not in small-rye, so its not instance of OriginTrackedMetadata - this makes it fall back to isReusable check.

NOTE: that OriginTrackedMetadata would also fail if those checks had reversed order.

**Comment by jmartisk** [comment]
Fix is in for 2.0
